### PR TITLE
HIP-584: Implement integration tests for token metadata & info functions for estimate_Gas

### LINKED ISSUE #5598: HIP-584: Provide integration tests for token info read-only precompiles
Provide integration tests for:

- GetFungibleTokenInfoPrecompile
-  GetTokenInfoPrecompile
- GetNonFungibleTokenInfoPrecompile

### LINKED ISSUE #5599: HIP-584: Provide integration tests for token metadata precompiles
Provide integration tests for:

- GetTokenDefaultFreezeStatus
- GetTokenDefaultKycStatus
- GetTokenTypePrecompile
- IsFrozenPrecompile
- IsKycPrecompile

### PR DESCRIPTION
**Pull request by MrValioBg**
**Description**:

Adds integration tests for the token metadata and token info readonly functions for estimate gas.

* In `ContractCallServicePrecompileTest` new test case for estimate_Gas is added. It uses the already existing enum `ContractReadFunctions`, thus providing tests for all the metadata token functions.
* New enum values are added for the token information functions.
* `estimateGasIterationThreshold` is increased to 7300, in order to support the test for the `getInformation` functions for NFTs.

**Related issue(s)**:

Fixes #5598
Fixes #5599 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java [hunk 1]
```diff
@@ -375,7 +383,13 @@ enum ContractReadFunctions {
...
                 new Object[] {NFT_ADDRESS_GET_KEY_WITH_DELEGATABLE_CONTRACT_ID, 64L},
                 new Object[] {false, Address.ZERO, new byte[0], new byte[0], CONTRACT_ADDRESS}),
-        GET_CUSTOM_FEES_FOR_TOKEN("getCustomFeesForToken", new Object[] {FUNGIBLE_TOKEN_ADDRESS}, new Object[] {});
+        GET_CUSTOM_FEES_FOR_TOKEN("getCustomFeesForToken", new Object[] {FUNGIBLE_TOKEN_ADDRESS}, new Object[] {}),
+        GET_FUNGIBLE_TOKEN_INFO(
+                "getInformationForFungibleToken", new Object[] {FUNGIBLE_TOKEN_ADDRESS}, new Object[] {}),
+        GET_NFT_INFO("getInformationForNonFungibleToken", new Object[] {NFT_ADDRESS, 1L}, new Object[] {}),
+        GET_INFORMATION_FOR_TOKEN_FUNGIBLE(
+                "getInformationForToken", new Object[] {FUNGIBLE_TOKEN_ADDRESS}, new Object[] {}),
+        GET_INFORMATION_FOR_TOKEN_NFT("getInformationForToken", new Object[] {NFT_ADDRESS}, new Object[] {});
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java:392]: Just to notify - we should consider refactoring this enum by adding complex return type objects for `getCustomFees` and `getTokenInfo` scenarios and by doing that we could directly use the enum values to validate the test and remove the `exclude` property. Then we could delete the stand-alone tests for `getCustomFees` and `getTokenInfo` and simplify the tests organisation and maintainability. 

I have logged a package in our backlog, covering this.

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM

**Review by bilyana-gospodinova — APPROVED** [review]
LGTM

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java (hunk 1 above)
Comment (MrValioBg) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java:392]: Great, thank you for taking care of this :) !

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public class MirrorNodeEvmProperties implements EvmProperties {
 
     @Getter
     @Positive
-    private long estimateGasIterationThreshold = 4200L;
+    private long estimateGasIterationThreshold = 7300L;
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:55]: Need to document the change in `docs/configuration.md`.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
LGTM besides conflict and doc request

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java (hunk 2 above)
Comment (MrValioBg) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:55]: Done :)

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**